src/main/java/controller/LadderGameController.java

```diff
@@ -0,0 +1,109 @@
+package controller;
+
+import domain.*;
+import view.InputView;
+import view.OutputView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LadderGameController {
+
+    private Users users;
+    private Ladder ladder;
+    private Results results;
+    private List<String> lastUserPositions;
+
+    public void makeLadderGame() {
+        generateUsers();
+        generateResults();
+        generateLadder();
+        printLadder();
+        calculateLadderGameResult();
+    }
+
+    public void printResult() {
+        try {
+            String name = InputView.readTargetUserName();
+            printResultWhenTargetIsNotAll(name);
+            printAllResult();
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+            printResult();
+        }
+    }
+
+    private void generateUsers() {
+        try {
+            users = new Users(InputView.readUserNames());
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+            generateUsers();
+        }
+    }
+
+    private void generateLadder() {
+        try {
+            ladder = new Ladder(InputView.readLadderHeight(), users.size(), new RandomLinkGenerator());
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+            generateLadder();
+        }
+    }
+
+    private void generateResults() {
+        try {
+            results = new Results(InputView.readResultNames(), users.size());
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+            generateResults();
+        }
+    }
+
+    private void printLadder() {
+        OutputView.printResultMessage();
+        OutputView.printUserNames(users.getUserNames());
+        OutputView.printLadder(formatPrimitiveLadder());
+        OutputView.printResultNames(results.getResultNames());
+    }
+
+    private List<List<Boolean>> formatPrimitiveLadder() {
+        List<List<Boolean>> primitiveLadder = new ArrayList<>();
+        List<Line> lines = ladder.getLadder();
+        for (Line line : lines) {
+            formatPrimitiveLine(primitiveLadder, line);
+        }
+        return primitiveLadder;
+    }
+
+    private void formatPrimitiveLine(final List<List<Boolean>> primitiveLadder, final Line line) {
+        List<Boolean> primitiveLine = new ArrayList<>();
+        for (Link link : line.getLinks()) {
+            primitiveLine.add(link.isLink());
+        }
+        primitiveLadder.add(primitiveLine);
+    }
+
+    private void calculateLadderGameResult() {
+        CalculateLadderGameResult calculator = new CalculateLadderGameResult();
+        lastUserPositions = calculator.passLadder(ladder.getLadder(), users.getUserNames());
+    }
+
+    private void printResultWhenTargetIsNotAll(String name) {
+        while (!name.equals("all")) {
```

Comment:
```suggestion
        while (!"all".equals(name)) {
```
우선 문자열 equals 비교를 할때는 변수보다 문자를 먼저 넣어주는것이 좋아요! 
이유는 한번 생각해보시면 좋을것 같아요

Reply:
잠시 찾아보니 앞의 `name` 변수에 값이 들어오지 않아 null이 된다면 equals를 호출하지 못하기 때문에 예외가 발생할 가능성이 있었습니다. 이미 값을 가지는 문자열이 확실히 있을 때는 그 문자열에 equals를 붙이는 것이 좋겠네요